### PR TITLE
Update Go to v1.17

### PR DESCRIPTION
**Pull request by relu**

### CODE REVIEW

**Comment by stefanprodan** [comment]
We can now move forward with this and update the source-controller packages to v0.20.0 which has Go 1.17 and controller-runtime 0.11.0

File: go.mod [hunk 1]
```diff
@@ -11,31 +11,116 @@ require (
 	github.com/fluxcd/image-automation-controller/api v0.18.0
 	// If you bump this, change REFLECTOR_VER in the Makefile to match
 	github.com/fluxcd/image-reflector-controller/api v0.14.0
```
Comment (relu) [inline go.mod:13]: We'll need to release this first.

File: go.mod [hunk 2]
```diff
@@ -11,31 +11,116 @@ require (
...
+	k8s.io/client-go v0.23.1
+	k8s.io/kube-openapi v0.0.0-20220104150652-0d48a347b383
+	sigs.k8s.io/controller-runtime v0.11.0
+	sigs.k8s.io/kustomize/kyaml v0.13.0
 )
 
-// Side-effect of depending on source-controller;
-// required by https://github.com/helm/helm/blob/v3.6.3/go.mod,
-// but overwritten with a newer version due to CVE-2017-11468.
-replace github.com/docker/distribution => github.com/docker/distribution v2.7.0-rc.0+incompatible
```
Comment (relu) [inline go.mod:41]: Does not seem to be required anymore.

File: pkg/update/testdata/setters/expected/kustomization.yaml [hunk 3]
```diff
@@ -2,8 +2,8 @@
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 resources:
-  - unimportant.yaml
```
Comment (relu) [inline pkg/update/testdata/setters/expected/kustomization.yaml:5]: With the new version of kyaml the indentation behavior changed. Are we ready for this?

File: go.mod (hunk 1 above)
Comment (stefanprodan) [reply go.mod]: This has been released.

File: pkg/update/testdata/setters/expected/kustomization.yaml (hunk 3 above)
Comment (stefanprodan) [reply pkg/update/testdata/setters/expected/kustomization.yaml:5]: Let's pin kyaml to the old version please.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @relu

**Pull request merged**